src/components/PaymentMethod.tsx

```diff
@@ -102,12 +105,37 @@ export const PaymentMethod: React.FunctionComponent<PaymentMethodProps> = props
               .map(_ => format(parseISO(_), "dd/MM/yyyy HH:mm"))
               .getOrElse("")}
           </div>
+          {props.walletItemInfo && PublicCreditCard.is(props.walletItemInfo) && (
+            <>
+              <div className="col-12 py-2"></div>
+              <div className="col-sm-2 font-weight-bold">
+                {t("Circuit Brand")}
+              </div>
+              <div className="col-sm-4">{props.walletItemInfo.brand}</div>
+              <div className="col-sm-2 font-weight-bold">{t("Masked PAN")}</div>
+              <div className="col-sm-4">*{props.walletItemInfo.masked_pan}</div>
+            </>
+          )}
           <div className="col-12 py-2"></div>
-          <div className="col-sm-2 font-weight-bold">HPAN</div>
+          <div className="col-sm-2 font-weight-bold">
+            HPAN
+            {props.walletItemInfo && (
+              <button
+                className="btn btn-sm btn-primary"
+                onClick={() => {
+                  popModal(props.walletItemInfo);
```

Comment:
We have a typings problem here. `popModal` argument should be `object` and `walletItemInfo` can be undefined. We could be change the signature of this function
```typescript
function popModal(data?: object): void {
    setModalcontent(data === undefined ? "undefined" : JSON.stringify(data, null, 3));
    setModalstate(!modalState);
  }
```